src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -163,4 +161,44 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
             s"from available source columns '${dfColumnNames.mkString(",")}'")
     }
   }
+
+  private def getIndexDataFrame(
+      spark: SparkSession,
+      df: DataFrame,
+      indexConfig: IndexConfig): (DataFrame, Seq[String], Seq[String]) = {
+    val (resolvedIndexedColumns, resolvedIncludedColumns) = resolveConfig(df, indexConfig)
+    val indexConfigColumns = resolvedIndexedColumns ++ resolvedIncludedColumns
+    val addLineage = spark.sessionState.conf
+      .getConfString(
+        IndexConstants.INDEX_LINEAGE_ENABLED,
+        IndexConstants.INDEX_LINEAGE_ENABLED_DEFAULT)
+      .toBoolean
+
+    val indexDF = if (addLineage) {
+      // Lineage is captured using two sets of columns:
+      // 1. DATA_FILE_NAME_COLUMN column contains source data filename for each index record.
+      // 2. If source data is partitioned, all partitioning key(s) are added to index schema.
+      val additionalPartitioningColumns = getPartitioningColumns(df).filter(
+        ResolverUtils.resolve(spark, _, indexConfigColumns).isEmpty)
+      val dataFrameColumns = indexConfigColumns ++ additionalPartitioningColumns
+      df.select(dataFrameColumns.head, dataFrameColumns.tail: _*)
+        .withColumn(IndexConstants.DATA_FILE_NAME_COLUMN, getFileName(input_file_name()))
+    } else {
+      df.select(indexConfigColumns.head, indexConfigColumns.tail: _*)
+    }
+
+    (indexDF, resolvedIndexedColumns, resolvedIncludedColumns)
+  }
+
+  private def getPartitioningColumns(df: DataFrame): Seq[String] = {
+    // Extract partitioning keys, if original data is partitioned.
+    val partitionSchema = df.queryExecution.optimizedPlan.collect {
+      case LogicalRelation(HadoopFsRelation(_, pSchema, _, _, _, _), _, _, _) => pSchema
+    }
+
+    partitionSchema.head.map(_.name)
+  }
+
+  private val getFileName: UserDefinedFunction = udf(
+    (fullFilePath: String) => FilenameUtils.getBaseName(fullFilePath))
```

Comment:
this will strip off extension as well. is that what we want?

Reply:
You are correct - I can not think of a case where dropping the extension could cause an issue. But adding it does not hurt either as most likely it wont cause an issue given that it will be repeated per file name and most likely Parquet's compression will take care of storing it efficiently. I can make sure we wont miss it if you have preference to capture it.

Reply:
Since these are "source" data, I could have "/tmp/1.csv", and "/tmp/1.parquet". If I tried to create an index on dataframe created by `spark.read.csv("/tmp/1.csv")`, would this be OK?

Reply:
I switched to `getName` to keep the extension.